django_rest_api/user/views.py

```diff
@@ -1,39 +1,48 @@
-from rest_framework import status, viewsets
+from rest_framework import status, views
 from rest_framework.response import Response
-from rest_framework.permissions import AllowAny
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.authentication import TokenAuthentication
 from rest_framework.authtoken.models import Token
 
 from .serializers import RegisterSerializer, LoginSerializer
-from .models import CustomUser
 
 
-class RegisterView(viewsets.ModelViewSet):
-    serializer_class = RegisterSerializer
+class RegisterView(views.APIView):
     permission_classes = [AllowAny]
-    queryset = CustomUser.objects.all()
-    
-    def create(self, request, *args, **kwargs):
+
+    def post(self, request, *args, **kwargs):
         serializer = RegisterSerializer(data=request.data)
         if serializer.is_valid():
             user = serializer.save()
             token, created = Token.objects.get_or_create(user=user)
             return Response({
+                    'id': user.pk,
                     'token': token.key,
-                    'user_id': user.pk,
-                    'email': user.email
+                    'username': user.username
                 }, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-
-class LoginView(viewsets.ModelViewSet):
-    serializer_class = LoginSerializer
+class LoginView(views.APIView):
     permission_classes = [AllowAny]
-    queryset = CustomUser.objects.all()
 
-    def create(self, request, *args, **kwargs):
+    def post(self, request, *args, **kwargs):
         serializer = LoginSerializer(data=request.data)
         if serializer.is_valid():
-            userData = serializer.save()
-            return Response(userData, status=status.HTTP_200_OK)
+            user = serializer.save()
```

Comment:
This could turn into `user = authenticate(serializer.validated_data["username"], serializer.validated_data["password"])` and you would save yourself from doing an extra DB call inside the serializer.